redbot/cogs/warnings/warnings.py

```diff
@@ -76,6 +84,49 @@ def __init__(self, bot: Red):
         else:
             await ctx.send(_("Custom reasons have been disabled."))
 
+    @warningset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
+        """Toggle warns being sent to the users DM's"""
+        guild = ctx.guild
+        toggle = await self.config.guild(guild).toggle_dm()
+        await self.config.guild(guild).toggle_dm.set(not toggle)
+        toggle = await self.config.guild(ctx.guild).toggle_dm()
+        if toggle == True:
+            await ctx.send("Warns will now be sent to users DM's")
+        else:
+            await ctx.send("Warns will not be sent to users DM's")
+
+    @warningset.command()
+    @commands.guild_only()
+    async def warnchannel(self, ctx: commands.Context, channel=None):
```

Comment:
You can use type casting for this to make it require a `discord.TextChannel`, so then you don't have to convert it later.